pymc/tests/distributions/test_transform.py

```diff
@@ -598,3 +615,31 @@ def test_discrete_trafo():
         with pytest.raises(ValueError) as err:
             pm.Binomial("a", n=5, p=0.5, transform="log")
         err.match("Transformations for discrete distributions")
+
+
+def test_transforms_ordered():
+    with pm.Model() as model:
+        pm.Normal(
+            "x_univariate",
+            mu=[-3, -1, 1, 2],
+            sigma=1,
+            size=(10, 4),
+            transform=tr.univariate_ordered,
+        )
+
+    log_prob = model.point_logps()
```

Comment:
To make this test a bit more readable I would suggest including a `x_ref = pm.Normal(..., shape=(4,))` in the model and comparing the elemwise logp of that is the same as each copy of the `(10, 4)` one. You can do that via `m.compile_logp(sum=False)({"x_1d_ordered__": np.zeros((4,)), x_2d_ordered__": np.zeros(10, 4)})` and then asserting closeness across the last axis.

Reply:
For the `sum_to_1` transform the logps are not the same. Is this expected? Even if the shape is the same like here:

```python3
def test_2d_univariate_sum_to_1():
    with pm.Model() as model:
        x_1d = pm.Normal(
            "x_1d",
            mu=[-3,-1,1,2],
            sigma=1,
            size=(10,4),
            transform=tr.univariate_sum_to_1,
        )
        x_2d = pm.Normal(
            "x_2d",
            mu=[-3, -1, 1, 2],
            sigma=1,
            size=(10, 4),
            transform=tr.univariate_sum_to_1,
        )

    log_p = model.compile_logp(sum=False)({"x_1d_sumto1__":np.ones((10,3))*0.25,"x_2d_sumto1__":np.zeros((10,3))*0.25})
    np.testing.assert_allclose(log_p[0],log_p[1])
```
Fails with:

```
./pymc/tests/distributions/test_transform.py::test_2d_univariate_sum_to_1 Failed: [undefined]AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 40 / 40 (100%)
Max absolute difference: 1.03125
Max relative difference: 0.72677567
 x: array([[-6.200189, -1.700189, -1.200189, -2.450189],
       [-6.200189, -1.700189, -1.200189, -2.450189],
       [-6.200189, -1.700189, -1.200189, -2.450189],...
 y: array([[-5.418939, -1.418939, -1.418939, -1.418939],
       [-5.418939, -1.418939, -1.418939, -1.418939],
       [-5.418939, -1.418939, -1.418939, -1.418939],...
```

Reply:
Ok funny. This is not the case if I use `np.zeros`

Reply:
PR looks good otherwise, should we investigate the `SumTo1`?

Reply:
Was this when comparing `np.zeros()`, vs `np.ones()`? I don't expect those to be equivalent...

Reply:
Oh boy :laughing: . Maybe it was just that. I thought I was comparing np.ones to ones. will check again.